connect/src/com/telenor/connect/ui/ConnectLoginButton.java

```diff
@@ -1,215 +1,148 @@
 package com.telenor.connect.ui;
 
-import android.content.ComponentName;
 import android.content.Context;
 import android.content.Intent;
-import android.content.pm.PackageManager;
-import android.net.Uri;
 import android.os.Build;
-import android.os.Bundle;
-import android.support.customtabs.CustomTabsCallback;
-import android.support.customtabs.CustomTabsClient;
-import android.support.customtabs.CustomTabsIntent;
-import android.support.customtabs.CustomTabsServiceConnection;
-import android.support.customtabs.CustomTabsSession;
-import android.text.TextUtils;
+import android.support.annotation.RequiresApi;
 import android.util.AttributeSet;
 import android.view.View;
+import android.widget.ProgressBar;
+import android.widget.RelativeLayout;
 
-import com.telenor.connect.BrowserType;
 import com.telenor.connect.ConnectSdk;
 import com.telenor.connect.R;
-import com.telenor.connect.sms.SmsBroadcastReceiver;
-import com.telenor.connect.sms.SmsHandler;
-import com.telenor.connect.sms.SmsPinParseUtil;
-import com.telenor.connect.sms.SmsRetrieverUtil;
-import com.telenor.connect.utils.ConnectUrlHelper;
-import com.telenor.connect.utils.CustomTabsHelper;
-import com.telenor.connect.utils.Validator;
-
-import java.lang.ref.WeakReference;
+import com.telenor.connect.headerenrichment.ShowLoadingCallback;
+import com.telenor.connect.id.Claims;
+
+import java.util.ArrayList;
 import java.util.Map;
 
-import static android.content.Intent.FLAG_ACTIVITY_NEW_TASK;
+public class ConnectLoginButton extends RelativeLayout implements AuthenticationButton {
+    ConnectLoginButtonWithoutProgressBar loginButton;
+    ProgressBar progressBar;
 
-public class ConnectLoginButton extends ConnectWebViewLoginButton {
+    public ConnectLoginButton(Context context) {
+        super(context);
+        init();
+    }
 
-    private static final Uri PRE_FETCH_URL
-            = Uri.parse(
-            ConnectUrlHelper
-                    .getConnectApiUrl()
-                    .newBuilder()
-                    .addPathSegment("id")
-                    .addPathSegment("android-sdk-prefetch-static-resources")
-                    .build()
-                    .uri()
-                    .toString()
-    );
+    private void init() {
+        inflate(getContext(), R.layout.com_telenor_connect_login_button_with_progress_bar, this);
```

Comment:
Here's the contrast again that I asked about in another class also. The class contains a button `ConnectLoginButtonWithoutProgressBar`, but here I see a parameter containing ...`_with_progress_bar`.

Reply:
Ref comment https://github.com/telenordigital/connect-android-sdk/pull/136#discussion_r240228262, is this more clear now?

Reply:
Yes, it's more clear. 
However, is it clear enough for other new readers of the code without explanation or thorough reading of the code? Consider if it makes sense to add a sentence as comment in the class. No biggie, drop it if it doesn't make sense.

Reply:
Is `ConnectCustomTabLoginButton` a better name than `ConnectLoginButtonWithoutProgressBar` perhaps?

Reply:
Yes, that makes sense to me and sounds like a good idea. I like that it relates to the class that it extends, namely `ConnectWebViewLoginButton`. Also, it avoids (or at least hides :) ) the confusion with the progress bar.

Reply:
Fixed: 2772159